filebeat/input/v2/input-cursor/input.go

```diff
@@ -109,12 +109,16 @@ func (inp *managedInput) Run(
 	defer cancel()
 	ctx.Cancelation = cancelCtx
 
+	// This context isn't directly used, so unregister any metric it might have
+	// registered.
+	ctx.UnregisterMetrics()
 	var grp unison.MultiErrGroup
 	for _, source := range inp.sources {
 		source := source
 		grp.Go(func() (err error) {
 			// refine per worker context
-			inpCtx := ctx
+			inpCtx := ctx.Clone()
```

Comment:
This seems unnecessary... `Context` is a struct, and the only thing in it that can't be shared is metrics, and the pipeline doesn't add any metrics registries before calling `Run` on the inputs, so there's no need for a new `Clone` method.

It's also not clear to me that we'd want different registries for these sources in the first place, do we? Is the intent that e.g. a single GCS input configured to read from multiple buckets will report each bucket under a different registry?

Reply:
I have the same questions as Fae.

Reply:
> It's also not clear to me that we'd want different registries for these sources in the first place, do we?

technically speaking, here we might end up with multiple inputs which each register its own metrics. From all I could see in the code only `ID`, never `IDWithoutName` is used to create input metrics. So I kept the behaviour, new context, new metrics registry. That's why I needed the clone method, otherwise I'd need to export the properties dealing with the registry.

Reply:
Worst case, it's always a 1-1 relationship, so nothing changes